cms/datasets/static/js/dataset-chooser.js

```diff
@@ -3,7 +3,8 @@
  * Automatically submits the search form when the published status filter changes
  */
 (function initDatasetChooser() {
-  const PUBLISHED_SELECTOR = 'select[name="published"]';
+  const PUBLISHED_SELECTOR =
+    'form[action="/admin/dataset_chooser/results/"] select[name="published"]';
```

Comment:
This won't work in the environment where we have overridden the admin URL from `/admin`, which is true for us.